### ISSUE TITLE
More detailed explanation for function prepareInterface()

### ISSUE THREAD

**Issue by docahrens** [jquery/learn.jquery.com#701]
The last example including the function prepareInterface() looks like a great example with lots to learn, but the explanation of the most difficult aspect of the example is lacking. In regard to prepareInterface(), it says: 
"The application also has a function (prepareInterface()) which returns a promise to complete animating our containers for both the news and reactions."

It would be very useful to see a much more detailed explanation of what is happening in prepareInterface() and how it is useful.  I am new to the idea of deferreds and promises, and unfortunately, it was not at all obvious to me what is happening in that method.  Seems like a tutorial would be an excellent place for such an explanation...

**Comment by agcolom** [comment]
Here's a link to the article: http://learn.jquery.com/code-organization/deferreds/jquery-deferreds/ for ref. @timmywil @dmethvin @jaubourg Could you please take a look and suggest how to go about this? Thanks

**Comment by dmethvin** [comment]
The example confuses me as well! It seems like you would not want to show the two sections until the data is filled, in which case the code should look like this?
```js
// Assume .news and .reactions are initially hidden
$.when(
    getLatestNews(),
    getLatestReactions()
).then(
    prepareInterface
)
...
```
The magic in `prepareInterface` is somewhat deep because it creates a new Deferred and uses `dfd.resolve` in the animation callback. Would this be clearer with a different name?
```js
function showAjaxedContent() {
    return $( ".news, .reactions" ).slideDown( 500, function() {
       $(this).addClass( "active" );
    }).promise();
}
```

**Comment by agcolom** [comment]
@docahrens would the suggestion by @dmethvin solve your issue? Thanks

**Comment by kswedberg** [comment]
Fixed in 9878874bfc350e44c2b8137cb3245e3ce4eb4a9b
